### PR TITLE
Make `<bool>.branch(...)` work

### LINKED ISSUE #4438: Make `<boolean>.branch(<b1>, <b2>)` work
In the JavaScript driver at least, we currently support
`r.branch(<boolean>, <b1>, <b2>)` but not `<boolean>.branch(<b1>, <b2>)`

I think we should support both syntaxes.

(checking Python and Ruby as well, even though I suspect that it might already be working in Ruby)

### PR DESCRIPTION
**Pull request by marshall007**
Fixes #4438.

### CODE REVIEW

File: drivers/python/rethinkdb/ast.py [hunk 1]
```diff
@@ -1483,7 +1486,7 @@ class Sync(RqlMethodQuery):
     st = 'sync'
 
 
-class Branch(RqlTopLevelQuery):
+class Branch(RqlMethodQuery):
```
Comment (danielmewes) [inline drivers/python/rethinkdb/ast.py]: I think this only determines how the term is pretty printed. I'd prefer keeping the old behavior here.

**Comment by danielmewes** [comment]
Thanks for the pull request @marshall007 !
I added one comment, but it looks good otherwise.

File: drivers/python/rethinkdb/ast.py (hunk 1 above)
Comment (marshall007) [inline drivers/python/rethinkdb/ast.py]: Ok, likewise in the JS driver should I revert [this line](https://github.com/rethinkdb/rethinkdb/pull/4442/files#diff-029874a8dbb2a1f773b7980319d494b5R978)?

Comment (danielmewes) [inline drivers/python/rethinkdb/ast.py]: Ah, yes please.

**Comment by danielmewes** [comment]
Sorry it took me a while to get back to this. Looks good, thank you @marshall007 !

**Pull request merged**